packages/git/src/node/dugite-git.ts

```diff
@@ -455,15 +455,22 @@ export class DugiteGit implements Git {
         );
     }
 
-    async show(repository: Repository, uri: string, options?: Git.Options.Show): Promise<string> {
+    async show(repository: Repository, uri: string, options?: Git.Options.Show): Promise<string | undefined> {
         const encoding = options ? options.encoding || 'utf8' : 'utf8';
         const commitish = this.getCommitish(options);
         const repositoryPath = this.getFsPath(repository);
         const path = this.getFsPath(uri);
-        if (encoding === 'binary') {
-            return (await getBlobContents(repositoryPath, commitish, path)).toString();
+        const errorCodes = new Set([0, 1, 128]); // 128 to handle if the file is not in Git.
```

Comment:
isn't 128 to general (in the code comment at least)?

Reply:
What would you expect? Just asking.

Reply:
I mean, I think it's right. but the comment might be misleading, as 128 is returned for all kind of errors, right?

Reply:
> What would you expect?

thirst thing I though reading this comment was, that this error case has a dedicated error code ;-) 
after looking at your changes to deguite-extra (https://github.com/theia-ide/dugite-extra/commit/e5263e2e486871186f560decc84584b82c0e886e#diff-ba3a11c6748e33b94d680245fe67866dR58), I understood that it's more like, if git crashes we return `undefined`.

Reply:
Kind of, there are a [few errors](https://github.com/desktop/dugite/blob/5c74239622fb896d3c3fa1a0f2d89ef0ce257981/lib/errors.ts#L2) that can be handled separately, but this is not among them.

Reply:
interesting hint! could we still parse the stdout provided by dugite to match that error?

e.g. 
```
> git show HEAD:unknown
fatal: Path 'unknown' does not exist in 'HEAD'
```

Reply:
that would be a more specific solution, but the more generic 128-code filter should work, too. I'm trying out.

Reply:
Sure, we can parse the error, but that implies, the error was already thrown. So with that approach, we would not suppress it. It worth a question from the dugite guys.

Reply:
do you mean that error here: https://github.com/theia-ide/dugite-extra/blob/e5263e2e486871186f560decc84584b82c0e886e/src/core/git.ts#L142 ?

Reply:
I see, I assumed it would be on that level: https://github.com/desktop/dugite/blob/master/examples/parsing-errors.ts#L9

Reply:
just to outline the option I've seen, after reading that error parsing example: instead of passing a set of acceptable codes, we could pass a second error code parser to `git` from `dugite-extra`, which would be chained after the existing one. that custom error parser could then test the `stderr`.

Reply:
but actually, there would be no more gain of that.

Reply:
I did a small experiment with different use-cases, it seems the error message can slightly vary based on the repository state:

```typescript
 describe('show', () => {
    it('exiting file', async () => {
      const testRepoPath = await initialize('desktop-show-existing')
      const filePath = Path.join(testRepoPath, 'file.txt')

      Fs.writeFileSync(filePath, 'some content', { encoding: 'utf8' })

      await GitProcess.exec(['add', '.'], testRepoPath)
      await GitProcess.exec(['commit', '-m', '"added a file"'], testRepoPath)

      const result = await GitProcess.exec(['show', ':file.txt'], testRepoPath)
      verify(result, r => {
        expect(r.exitCode).to.be.equal(0)
        expect(r.stdout.trim()).to.be.equal('some content')
      })
    })
    it('missing from index', async () => {
      const testRepoPath = await initialize('desktop-show-missing-index')

      const result = await GitProcess.exec(['show', ':missing.txt'], testRepoPath)
      verify(result, r => {
        expect(r.exitCode).to.be.equal(128)
        expect(
          r.stderr
            .trim()
            .endsWith(`Path 'missing.txt' does not exist (neither on disk nor in the index).`)
        ).to.be.true
      })
    })
    it('missing from HEAD', async () => {
      const testRepoPath = await initialize('desktop-show-missing-head')

      const filePath = Path.join(testRepoPath, 'file.txt')

      Fs.writeFileSync(filePath, 'some content', { encoding: 'utf8' })

      await GitProcess.exec(['add', '.'], testRepoPath)
      await GitProcess.exec(['commit', '-m', '"added a file"'], testRepoPath)

      const result = await GitProcess.exec(['show', 'HEAD:missing.txt'], testRepoPath)
      verify(result, r => {
        expect(r.exitCode).to.be.equal(128)
        expect(r.stderr.trim().endsWith(`Path 'missing.txt' does not exist in 'HEAD'`)).to.be.true
      })
    })
    it('missing from HEAD - no HEAD', async () => {
      const testRepoPath = await initialize('desktop-show-missing-head')

      const result = await GitProcess.exec(['show', 'HEAD:missing.txt'], testRepoPath)
      verify(result, r => {
        expect(r.exitCode).to.be.equal(128)
        expect(r.stderr.trim().endsWith(`Invalid object name 'HEAD'.`)).to.be.true
      })
    })
    it('-- missing', async () => {
      const testRepoPath = await initialize('desktop-show-missing-head')

      const filePath = Path.join(testRepoPath, 'file.txt')

      Fs.writeFileSync(filePath, 'some content', { encoding: 'utf8' })

      await GitProcess.exec(['add', '.'], testRepoPath)
      await GitProcess.exec(['commit', '-m', '"added a file"'], testRepoPath)

      const result = await GitProcess.exec(['show', '--', 'missing.txt'], testRepoPath)
      verify(result, r => {
        expect(r.exitCode).to.be.equal(0)
        expect(r.stdout.trim()).to.be.empty
      })
    })
    it('-- missing - outside', async () => {
      const testRepoPath = await initialize('desktop-show-missing-head')

      const filePath = Path.join(testRepoPath, 'file.txt')

      Fs.writeFileSync(filePath, 'some content', { encoding: 'utf8' })

      await GitProcess.exec(['add', '.'], testRepoPath)
      await GitProcess.exec(['commit', '-m', '"added a file"'], testRepoPath)

      const result = await GitProcess.exec(['show', '--', '/missing.txt'], testRepoPath)
      verify(result, r => {
        expect(r.exitCode).to.be.equal(128)
        expect(r.stderr.trim().endsWith('is outside repository')).to.be.true
      })
    })
  })
```

Reply:
I would do the followings:
 - Go with `128` as is now.
 - If the result is `undefined` we check the `stderr` **in the Theia code**, and if that contains information about the missing path, we ignore it. Otherwise, we can throw an error.

With this, we ignore the missing path in Theia, we keep the `dugite-extra` code as close to the GitHub Desktop code as possible. But we still throw an error if we suspect something unexpected error message from Git.